lib/Common/CommonDefines.h

```diff
@@ -120,6 +120,21 @@
 
 // GC features
 
+#define BUCKETIZE_MEDIUM_ALLOCATIONS 1              // *** TODO: Won't build if disabled currently
+#define SMALLBLOCK_MEDIUM_ALLOC 1                   // *** TODO: Won't build if disabled currently
+#define LARGEHEAPBLOCK_ENCODING 1                   // Large heap block metadata encoding
+#define RECYCLER_WRITE_BARRIER                      // Write Barrier support
+#define IDLE_DECOMMIT_ENABLED 1                     // Idle Decommit
+#define RECYCLER_PAGE_HEAP                          // PageHeap support
+
+#ifdef RECYCLER_WRITE_BARRIER
+#if defined(__clang__) 
```

Comment:
>__clang__ [](start = 12, length = 9)

I think some `__clang__` uses should be `_WIN32`. We can use clang on Windows with hardware write barrier.